src/main/java/in/projecteka/consentmanager/user/SessionService.java

```diff
@@ -20,13 +21,19 @@
     private final TokenService tokenService;
     private final CacheAdapter<String, String> blacklistedTokens;
     private final Logger logger = LoggerFactory.getLogger(SessionService.class);
+    private final LockedUserService lockedUserService;
 
     public Mono<Session> forNew(SessionRequest request) {
         if (StringUtils.isEmpty(request.getUsername()) || StringUtils.isEmpty(request.getPassword()))
             return Mono.error(ClientError.unAuthorizedRequest());
         return tokenService.tokenForUser(request.getUsername(), request.getPassword())
                 .doOnError(error -> logger.error(error.getMessage(), error))
-                .onErrorResume(error -> Mono.error(ClientError.unAuthorizedRequest()));
+                .onErrorResume(InvalidUserNameException.class, error -> Mono.error(ClientError.invalidUserName()))
+                .onErrorResume(error -> lockedUserService.userFor(request.getUsername())
+                        .switchIfEmpty(
```

Comment:
This can be empty if the getLockedUser from repository returned no results. How can the user be unauthorized if it is empty?

Reply:
It doesn’t matter whether it is empty or. if the password or username is wrong then it'll be unauthorized

Reply:
for the first time invalid attempt no user will be there in DB